### DISCUSSION TITLE
Perbedaan antara RMD dengan Knit

### DISCUSSION THREAD

**Discussion by yohanespm77** [teamalgoritma/community#688] [Q&A] [answered]
Saya melakukan knit pada rmd saya, namun hasilnya berbeda ketika di knit.

# Di RMD
<img width="545" alt="image" src="https://github.com/teamalgoritma/community/assets/106010284/2a371899-a195-4b59-9bfb-22cf12cc9be9">

# Pada saat di Knit
<img width="600" alt="image" src="https://github.com/teamalgoritma/community/assets/106010284/0fa31e5b-b508-463a-a57a-87effba36ce0">

Mohon bantuannya kak, terima kasih 🙏

**Comment by IrfanChairurrachman** [comment, marked as answer]
Dear @yohanespm77 ,

Hal ini kemungkinan karena data yang digunakan untuk clustering kmeans pada saat di-knit berbeda dengan saat dijalankan di RMD, ini dapat terjadi ketika di **pemrosesan data terdapat efek random** (seperti sampling data untuk train-test splitting) sehingga setiap dijalankan hasilnya akan berbeda dan mempengaruhi hasil-hasil di bawahnya. 

Anda dapat melakukan inspeksi dengan `head()` atau `tail()` sebelum dilakukan `fviz_nbclust` untuk membandingkan data setiap row sama ketika hasil knit dan di rmd. Jika setiap row datanya berbeda Anda dapat melakukan **set seed** dengan cara di bawah ini 
```
RNGkind(sample.kind = "Rounding")
set.seed(100)
```

Silahkan **set seed** chunk-chunck yang memiliki efek random untuk **mengunci efek kerandoman** dari pemrosesan data yang dilakukan sehingga setiap dijalankan hasilnya akan selalu sama (termasuk ketika di-knit).

Sekian, semoga membantu

**Reply by yohanespm77** [reply]
Sudah di set seed kak, tapi masih tidak sama outputnya

**Reply by IrfanChairurrachman** [reply]
Apakah sudah dicoba Pak untuk melihat isi datanya menggunakan `head` atau `tail` sebelum dilakukan `fviz_nbclust`? Apakah isi datanya berbeda atau tetap sama?

**Reply by yohanespm77** [reply]
Isi datanya sama mas, hanya visualisasi `fviz_bvclust`nya yang berbeda

**Reply by IrfanChairurrachman** [reply]
Mungkin boleh Pak dilampirkan screenshot untuk melihat data tepat sebelum dilakukan `fviz_nbclust` baik di Rmd dan hasil setelah knit. Contohnya seperti berikut:

### Rmd
![image](https://github.com/teamalgoritma/community/assets/56690035/3408d691-33e4-4084-a289-ff206fbe8fa5)

###  Knit
![image](https://github.com/teamalgoritma/community/assets/56690035/a1fd305c-d3a7-44d9-9599-570375353fcf)
